luarules/gadgets/game_autocolors.lua

```diff
@@ -736,5 +788,14 @@ else	-- UNSYNCED
 			Spring.SetConfigInt("SimpleTeamColors_Reset", 0)
 		end
 	end
-end
 
+	function gadget:PlayerChanged(playerID)
+		if playerID ~= myPlayerID then
+			return
+		end
+
+		mySpecState = Spring.GetSpectatingState()
+
+		Spring.SetConfigInt("UpdateTeamColors", 1)
+	end
```

Comment:
And this is the third place that might be breaking team colors.

Reply:
Thank you, I'll take a look at it

Reply:
It's possible the updateteamcolors function is not idempotent